### ISSUE TITLE
No automatic reload after restart

### ISSUE THREAD

**Issue by catterz** [EpiCanard/GlobalMarketChest#74]
Issue goal: Bug
Platform: PaperMC 1.13.2
Plugin version:GlobalMarketChest v1.2.0

**Description**
After restarting my server, GMC does not automatically load the config again until I manually run /gmc reload

There are no errors in the logs anywhere and everything runs perfectly after calling /gmc reload. Could it be a possible conflict? /gmc is used in EssentialsX for "Gamemode Change". The EssentialsX plugin is loading after GlobalMarketChest too, so possibly is overwriting the command? I'm not sure how plugin code works, so just a guess!

**Comment by EpiCanard** [comment]
If you use the command /Globalmarketchestt and not the alias it works or not ?

I'm sure the config file load, if it wasn't the case the plugin may be disabled. Because I need the config file for many use that can generate errors if they don't have config.

**Comment by catterz** [comment]
The command works, but there are no shops listed if I run `/gmc list` until I run `/gmc reload`

So I guess the plugin is loaded fine, but the data hasn't loaded?

Also, before calling `/gmc reload` the chest and sign are just a normal chest/sign and do not open a shop.

**Comment by EpiCanard** [comment]
Maybe the server hasn't fully reload ? If you can execute the command reload it means the plugin is enabled. But if it is enabled you should have database connection and your shops. That make no sense.

Can you send your logs ?

**Comment by catterz** [comment]
[latest.log](https://github.com/EpiCanard/GlobalMarketChest/files/3200102/latest.log)
It's quite large and there are a fair amount of plugins.

**Comment by EpiCanard** [comment]
I haven't explications ^^"
All seems okay.

How are you doing your server reload ? With`/reload` ?
The first time you start your serveur(stop and start) it works or not ?

I will try to do more tests with paperMC maybe it's having cache for config file.

**Comment by catterz** [comment]
I'm hosting with server.pro and use their control panel to initiate a server restart. I have tried stopping and starting 2 or 3 times and the results are the same each time unfortunately.

**Comment by EpiCanard** [comment]
It's really strange. I guess it can be caused by Paper because I have no issues with spigot.
I will try to reproduce the bug.

**Comment by catterz** [comment]
I wouldn't be surprised if it was one of my dozens of plugins, but thanks for looking!

**Comment by EpiCanard** [comment]
Can you send me a list of all your plugins ? Maybe one interact with others.

**Comment by catterz** [comment]
Sure:

AdvancedBan-2.1.5-RELEASE
CoreProtect-2.16.3
CustomHeads
Discord Bot API - V2.8
Discord Minecraft Hook - V4.0
EditableSign
EssentialsX-2.16.1.0
EssentialsXChat-2.16.1.0
EssentialsXSpawn-2.16.1.0
GlobalMarketChest-1.2.0
Holograms-2.6-SNAPSHOT
LuckPerms-Bukkit-4.4.1
Multiverse-Core-3.0.0-SNAPSHOT
Multiverse-Inventories-3.0.0-SNAPSHOT
Multiverse-NetherPortals-3.0.0-SNAPSHOT
Multiverse-Portals-3.0.0-SNAPSHOT
PlotSquared-Bukkit-4.226
RifleChairs
SalaryManager
Shop
Vault
VotingPlugin
nuvotifier-2.5.2
uSkyBlock-20190328-1-ec2a672
worldedit-bukkit-7.0.0-beta-05
worldguard-legacy-7.0.0-beta-03

**Comment by EpiCanard** [comment]
Thank you

**Comment by EpiCanard** [comment]
When you execute `/gmc list` there is no error inside console ?

**Comment by catterz** [comment]
Nothing:

`[INFO] acatterz issued server command: /gmc list`

![image](https://user-images.githubusercontent.com/25639922/58074820-0e829980-7b9e-11e9-95a3-4b9e2f9f0cc4.png)

And then I run /gmc reload:

`[INFO] acatterz issued server command: /gmc reload`

![image](https://user-images.githubusercontent.com/25639922/58074830-18a49800-7b9e-11e9-8eca-e1ad5da9be8e.png)

And then /gmc list again:

`acatterz issued server command: /gmc list`

![image](https://user-images.githubusercontent.com/25639922/58074866-2eb25880-7b9e-11e9-85d3-35886c072cf5.png)

Screenshot of the console. There is no output after running any command:

![image](https://user-images.githubusercontent.com/25639922/58074993-76d17b00-7b9e-11e9-99ff-053c1e1c34ff.png)

**Comment by EpiCanard** [comment]
Wtf.
Hoping I will find something because I hate trying to fix that kind of bug.

**Comment by catterz** [comment]
Let me know if there is anything I can do to help. I have to go to work soon! Don't stress over it too much as it is possibly just a conflict. I can work around it by running the command on a restart at least.

In case it helps, here is my config file (I have redacted the username and password for MySQL).
[config.txt](https://github.com/EpiCanard/GlobalMarketChest/files/3201181/config.txt)

Mostly everything is left as default.

**Comment by EpiCanard** [comment]
I will have to go to work too, I can't try to fix it before this evening.

I don't think it is cause by the database connection because you have an error when it can't connect to it.

**Comment by EpiCanard** [comment]
I see that you use an old version of paperMC. You use the version 493 from 09-01-2019, can you try the last build 626 ?

If it doesn't work can you send me a copy of your plugins folder, without private information like password, etc...
If you prefer you can send it to me in private message on discord or spigot.

**Comment by catterz** [comment]
I don't have control over the Paper build unfortunately as that part is tied to server.pro hosting. I will send you a copy of my plugins folder as requested. Should be within 24-48 hours as I'm not going to get a good chance to go through all of the configs until Thursday realistically. Thanks for your help!

**Comment by EpiCanard** [comment]
Ok, I will wait.
I will keep you informed of the progress when I will have your plugin folder.

**Comment by catterz** [comment]
[DropBox Link to plugins.zip](https://www.dropbox.com/s/ydmr26361jelgbo/plugins.zip?dl=0)

The file was quite large so I removed my CoreProtect database to save some space is it was over 400 MB. Also, you might notice a few extra plugins since the list above, so you can ignore those if it's easier.

Changes I made to the zip before uploading:

- Removed MySQL connection details from GlobalMarketChest\config.yml
- Removed tokens from both Minecraft Hook folders
- Removed public and private keys from Votifier

**Comment by catterz** [comment]
(Please can you let me know when you have downloaded the file and I will revoke access. Thanks!)

**Comment by EpiCanard** [comment]
It's ok you can revoke access :)

**Comment by EpiCanard** [comment]
I tried with the permission group **default** that can use _globalmarketchest_ commands and when I execute **/gmc list** all works as expect, I have all the shops well listed even the first time.

If it is a plugin conflict I have only these plugins disabled :
- Votifier
- DiscordBot

Do you have change server configs ? Like _server.properties_ or _paper.yml_ / _spigot.yml_ ?

**Comment by catterz** [comment]
[Server Files.zip](https://github.com/EpiCanard/GlobalMarketChest/files/3214261/Server.Files.zip)

I may have made some tweaks, but I don't remember exactly what. All the relevant files should be in the zip.

**Comment by EpiCanard** [comment]
FOUND !!!!! [Yeah !!!](https://media.giphy.com/media/yoJC2GnSClbPOkV0eA/giphy.gif)

In your config file `server.properties`, the config variable `level-name=Plots\!` seems to affect this issue.
Why ? I don't know. 
Maybe it is the world config or maybe it is the world name. It is better to have a world name only with lowercase letters and underscore.
Here you have an `!` inside the name and it can cause unwanted issues I think.

**Comment by catterz** [comment]
Sorry, I didn't get a notification here so a bit later replying! Honestly I'm not sure why it was created with an exclamation mark in the first place (Picard).

I've since moved my spawn world so I've updated the setting anyway to "spawn2" so should be compliant. In the meantime I had worked around the issue by using Skript to schedule a /gmc reload 10 seconds after server start.

Thanks for your support.
